### PR TITLE
[move-ide] Extended on-hover for struct types

### PR DESCRIPTION
**Pull request by awelc**
## Description 

This PR extends on-hover functionality for struct types - when hovering over the type, instead of just seeing the type name, what is now displayed is the whole structure of the type (including fields). In addition to being more informative, this is also more in line with what rust-analyzer provides.

## Test Plan 

Test have been adjusted to reflect this change.

### CODE REVIEW

File: external-crates/move/crates/move-analyzer/src/symbols.rs [hunk 1]
```diff
@@ -356,7 +421,6 @@ fn addr_to_ide_string(addr: &Address) -> String {
         Address::NamedUnassigned(name) => format!("{}", name),
     }
 }
-
```
Comment (rvantonder) [inline external-crates/move/crates/move-analyzer/src/symbols.rs:359]: flagging empty line (?)

**Review by rvantonder — APPROVED** [review]
Looks good!

>  instead of just seeing the type name, what is now displayed is the whole structure of the type (including fields)

Just a thought, in some IDE contexts I've seen it useful to have a more terse description of types (like the one-liner we had before). For example, sometimes on hover I might only see something like this in Rust:

> let mut config: BuildConfig // size = 104 (0x68), align = 0x8   

Which is probably configurable, and depending on the context / screen space I might want the full definition or not. Not sure if this is a rust analyzer option though or some plugin middleware. IIRC VSCode probably handles screen space with a scrollable dialog (and on balance having the struct defs is probably more useful). 

TL;DR just wondering if we want to consider building in optionality for this display (can't remember if LSP supports something like that directly though), or whether big structs pose display issues for editors. Again, just thoughts!

**Comment by awelc** [comment]
> Just a thought, in some IDE contexts I've seen it useful to have a more terse description of types (like the one-liner we had before). For example, sometimes on hover I might only see something like this in Rust:
> 
> ```rust
> let mut config: BuildConfig // size = 104 (0x68), align = 0x8
> ```

I modeled the current behavior on rust-analyzer, at least on its default behavior (I am not sure if it's configurable, but either way we could certainly consider it). This default behavior is consistent with your observation:
- when hovering over variable/parameter definition or a field access, the description shown is the terse one (just the name of the struct type)
- when hovering over actual name of the struct type, the description shown is verbose

Granted, our terse version is still a bit different (and simpler) - I am working on bringing it more in line with rust-analyzer in the subsequent PR.

Here is our current terse version when hovering over parameter `s` (which was all we were showing ever before the current PR):
image

Here is our current verbose version when hovering over `DocumentedStruct` parameter type:

image

**Comment by rvantonder** [comment]
> when hovering over actual name of the struct type, the description shown is verbose

Ah, got it, thanks for clarifying and the examples, looks great.
